src/bgw/job_stat.c

```diff
@@ -161,11 +161,21 @@ typedef struct
 static TimestampTz
 calculate_next_start_on_success(TimestampTz last_finish, BgwJob *job)
 {
-	/* TODO: add randomness here? Do we need a range or just a percent? */
-	TimestampTz ts =
-		DatumGetTimestampTz(DirectFunctionCall2(timestamptz_pl_interval,
-												TimestampTzGetDatum(last_finish),
-												IntervalPGetDatum(&job->fd.schedule_interval)));
+	TimestampTz ts;
```

Comment:
```suggestion
	volatile TimestampTz ts;
```
if we're going to change it in both branches of the `try/catch` this is required.